### DISCUSSION TITLE
What's the best way to get unique values without fetching the whole collection?

### DISCUSSION THREAD

**Discussion by rooselle** [statamic/cms#11126] [Q&A]
Hi,

I have trouble wrapping my head around the Statamic query system and I'm getting a bit frustrated that I cannot use all the usual Eloquent methods. :innocent: That's learning, I guess :nerd_face:

I have the following code that works:
```
<select name="storage" class="select select-bordered">
     <option value="">Tous les rangements</option>
    {{ collection :from="mount" as="all_fabrics" scope="all" sort="storage:asc" }}
        {{ all_fabrics unique="storage" }}
            <option value="{{ storage }}" {{ if {get:storage} == storage }}selected{{ /if }}>{{ storage }}</option>
        {{ /all_fabrics }}
    {{ /collection }}
</select>
```

However, it's seems a bit overkill to me to fetch all of the fabrics to display in the end only the ~15 different storages. Especially since I have several other `<select>` on the same page for other fabric attributes. In Eloquent, I would have done something like `Fabric::select('storage')->distinct()->get()`. How can I achieve the same with Statamic Stache?

Thanks for your help!

**Comment by robdekort** [comment]
Perhaps you could restructure your setup to use taxonomies for storages instead? You could then use the taxonomy tag with the `min_count` parameter.

**Reply by rooselle** [reply]
Thanks for the tip! Following this logic, I should then also make Taxonomies `pattern`, `owner`, `state` (whether it's new, used, etc.)... which would only be used by the Collection `fabrics`. I wonder if it's a good way to go?

**Reply by robdekort** [reply]
I'm not sure what those do, but yeah. That could probably work. I usually use taxonomies when they describe something reusable about a collection. 

There's a few downsides to taxonomies currently however when you use multisite or when you want them to have custom routes. As long as you don't need that, taxonomies can be really handy to work with.

**Reply by rooselle** [reply]
They're other properties of a fabric (my app lists fabrics). So they're indeed reusable inside the collection, and I don't need custom routes for those. So I'll try that, thank you again!

**Reply by robdekort** [reply]
Welcome, good luck!
